backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestComputationCommand.java

```diff
@@ -148,36 +148,35 @@ private InitialInterestAllocationResult createInterestRecords(
 			@NonNull final FlatrateTermId contractId)
 	{
 		final Iterator<ModularContractLogEntry> shippingNotificationIterator = streamShippingNotificationLogEntries(request, contractId).iterator();
-		final Iterator<ModularContractLogEntry> interimInvoiceIterator = streamInterimInvoiceLogEntries(contractId, request).iterator();
+		final Iterator<ModularContractLogEntry>  interimContractIterator = streamInterimContractLogEntries(contractId, request).iterator();
 		final ModularContractSettings settings = modularContractService.getModularSettingsForContract(contractId);
 
 		BigDecimal totalInterestScore = BigDecimal.ZERO;
+		final boolean interimInvoiceExists = interimInvoiceLogExists(contractId, request);
 
-		InvoiceAllocations.AllocationItem currentShippingNotification = null;
-		while (interimInvoiceIterator.hasNext())
+		InterimContractAllocations.AllocationItem currentShippingNotification = null;
+		while(interimContractIterator.hasNext())
 		{
-			final InvoiceAllocations currentInvoiceAllocations = initInvoiceAllocations(request,
-					settings.getAdditionalInterestDays(),
-					interimInvoiceIterator.next());
-			while (currentInvoiceAllocations.openAmountSignum() > 0)
+			final InterimContractAllocations currentInterimContractAllocations = initInterimContractAllocations(request,
+																												settings.getAdditionalInterestDays(),
+																												interimContractIterator.next());
+
+			while (currentInterimContractAllocations.openAmountSignum() > 0)
 			{
-				while (!currentInvoiceAllocations.canAllocate(currentShippingNotification) && shippingNotificationIterator.hasNext())
+				while (!currentInterimContractAllocations.canAllocate(currentShippingNotification) && shippingNotificationIterator.hasNext())
 				{
```

Comment:
Do we still need this check?

Reply:
How openAmount is computed now?

Reply:
I tried to keep it like before, only with the amount from contract instead of interim invoice. See https://github.com/metasfresh/metasfresh/pull/18378/files#diff-468149c0aac8a65ac7d3c51ca49bd620834929280de1d8b58d6a6013cb266454R368